### PR TITLE
Refactor doctor

### LINKED ISSUE #12706: [Bug] doctor: crash if not able to open doctor.log
Version v1.12.4

`sudo -u gitea gitea doctor --all` executed will just crash with:

```
panic: Failed to create sublogger (doctor): open doctor.log: permission denied

goroutine 1 [running]:
code.gitea.io/gitea/modules/log.NewLogger(0x3e8, 0x20d618a, 0x6, 0x20be030, 0x4, 0xc000192370, 0x42, 0x1fa9d20)
	/home/user/aports/community/gitea/src/src/code.gitea.io/gitea/modules/log/log.go:48 +0x1ad
code.gitea.io/gitea/cmd.runDoctor(0xc0003c8840, 0x0, 0xc0003c86e0)
	/home/user/aports/community/gitea/src/src/code.gitea.io/gitea/cmd/doctor.go:149 +0x1aa
github.com/urfave/cli.HandleAction(0x1d49880, 0x2283168, 0xc0003c8840, 0xc00035ba00, 0x0)
	/home/user/aports/community/gitea/src/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/app.go:490 +0x82
github.com/urfave/cli.Command.Run(0x20d618a, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x218d96a, 0x11, 0x0, ...)
	/home/user/aports/community/gitea/src/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/command.go:210 +0x9fb
github.com/urfave/cli.(*App).Run(0xc001222340, 0xc0000a2150, 0x3, 0x3, 0x0, 0x0)
	/home/user/aports/community/gitea/src/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/app.go:255 +0x768
main.main()
	/home/user/aports/community/gitea/src/src/code.gitea.io/gitea/main.go:109 +0x972
```

## I expect:
... that doctor tell me what I have to change to be able to execute it

### PR DESCRIPTION
**Pull request by zeripath**
This PR refactors the doctor code to move it to its own module. It changes the functions to use a passed in logger instead of using a results array. 

Fix #12706

### CODE REVIEW

**Comment by 6543** [comment]
so fare so good - I think with this refactor maintaining doctor is easyer ...

**Comment by zeripath** [comment]
They're not quite unrelated ... we need them for this

File: modules/doctor/misc.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			_, err := git.NewCommand("config", "receive.advertisePushOptions", "true").RunInDir(r.Path)
+			return err
+		}
+
+		value, err := git.NewCommand("config", "receive.advertisePushOptions").RunInDir(r.Path)
+		if err != nil {
+			return err
+		}
+
+		result, valid := git.ParseBool(value)
```
Comment (6543) [inline modules/doctor/misc.go]:
```suggestion
		result, valid := git.ParseBool(strings.TrimSpace(value))
```

**Comment by lunny** [comment]
It's better to move all codes about logger to another PR and this one depends that one.

**Comment by zeripath** [comment]
Then we will need to merge those changes first.

The logger changes are entirely required for this to work

File: modules/doctor/misc.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package doctor
+
+import (
+	"fmt"
+	"os/exec"
```
Comment (6543) [inline modules/doctor/misc.go:9]:
```suggestion
	"os/exec"
	"strings"
```

Comment (6543) [reply modules/doctor/misc.go:9]: sorry missed that :(

**Comment by 6543** [comment]
@zeripath do this solve  #12706?

**Comment by zeripath** [comment]
Probably it does solve it yes

**Review by 6543 — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**